### PR TITLE
RDMA bind local address

### PR DESCRIPTION
**Pull request by pizhenwei**
Hi @michael-grunder @bjosv 

Each patch has independent functionality, also been tested by CI and local commands, so request no-squash merge for this PR.

### CODE REVIEW

File: src/rdma.c [hunk 1]
```diff
@@ -857,6 +858,18 @@ static int valkeyContextConnectRdma(valkeyContext *c, const valkeyOptions *optio
         }
     }
 
+    if (!source_addr) {
```
[the remark this answers is not in the document]
Comment (bjosv) [reply src/rdma.c:861]: ..to avoid mixing styles, since we have `c->tcp.source_addr == NULL` below

**Review by bjosv — COMMENTED** [review]
Just some nits

**Review by bjosv — APPROVED** [review]

**Review by michael-grunder — APPROVED** [review]
The lines like 

```c
snprintf(_port, 6, "%d", port);
```

Could be 
```c
snprintf(_port, sizeof(_port), "%d", port);
```

But I don't think it matters as `sizeof("65536")` isn't changing any time soon :smile:

**Comment by pizhenwei** [comment]
Hi @michael-grunder 
Push a new version as you suggested, thanks!

**Comment by bjosv** [comment]
Looks good, thanks!

**Pull request merged**